extensions/wikia/Flags/maintenance/MoveNotices.php

```diff
@@ -0,0 +1,465 @@
+		$this->prepareLogFile( $csv );
+
+		if ( !$this->logFile ) {
+			$this->output( "Cannot create log file.\n" );
+			return;
+		}
+
+		$csvFile = fopen( $csv, 'r' );
+		$csvData = [];
+
+		if ( !$csvFile ) {
+			$this->output( "Cannot read file: $csv" );
+			return;
+		}
+
+		// Get data from scv file
+		while( ( $data = fgetcsv( $csvFile ) ) !== false ) {
+			$csvData[] = $data;
+		}
+
+		fclose( $csvFile );
+
+		$this->output( "Start processing\n" );
+
+		$flagModel = new Flag();
+		$flagTypeModel = new FlagType();
+		$flagsExtractor = new FlagsExtractor();
+
+		foreach( $csvData as $data ) {
+			$this->log = '';
+			$flagId = null;
+
+			$error = $this->validateCSVData( $data );
+
+			if ( $error ) {
+				fwrite($this->logFile, $this->log);
+				$this->output( $this->log );
+				continue;
+			}
+
+			$this->templateName = $data[0];
+
+			if ( !$list ) {
+				// Prepare data to add flag type
+				$flagType = $this->prepareDataForFlagType( $flagTypeModel, $data );
+
+				if ( !($this->flagId = $this->addFlagType( $flagTypeModel, $flagType ) ) ) {
+					continue;
+				}
+
+				$this->log = "Adding flag type: " . json_encode( $flagType ) ."\n";
+			}
+
+			$this->log .= "Start processing template: $this->templateName \n";
+
+			$title = Title::newFromText('Template:' . $this->templateName);
+
+			$rows = $this->showIndirectLinks( 0, $title, 0 );
+
+			if ( empty( $rows ) ) {
+				$this->log .= "[WARNING] This template is not used \n";
+				$this->log .= "================================================== \n\n\n";
+				fwrite( $this->logFile, $this->log );
+				$this->output( $this->log );
+				continue;
+			}
+
+			if ( !is_null( $section )  ) {
+				$this->log .= "Searching in section $section\n";
+			} else {
+				$this->log .= "Searching in all article content\n";
+			}
+
+			fwrite( $this->logFile, $this->log );
+			$this->output( $this->log );
+
+			foreach ( $rows as $row ) {
+				$this->log = '';
+				$this->pageId = $row->page_id;
+
+				$page = Title::makeTitle( $row->page_namespace, $row->page_title );
+				$pageName = $page->getPrefixedText();
+				$article = Article::newFromID( $this->pageId );
+				$content = $article->getContent();
+
+				if ( !is_null( $section )  ) {
+					$content = $wgParser->getSection($content, $section);
+				}
+
+				$this->log .= "Looking for template on $pageName [" . $this->pageId . "]\n";
+
+				$flagsExtractor->init( $content, $this->templateName);
+				$templates = $flagsExtractor->getAllTemplates();
+
+				$size = sizeof( $templates );
+
+				if ( !$size ) {
+					$this->log .= "[WARNING] No templates found on page $pageName\n";
+				} elseif ( $size > 1 ) {
+					$this->log .= "[WARNING] Found more than one ($size) template $this->templateName on page $pageName\n";
+				}
+
+				if ( $size ) {
+					$this->logTemplatesInfo( $templates );
+
+					$flagsToPages = $this->prepareDataForFlagsToPage( $templates[0]['params'] );
+
+					if ( !$list ) {
+						$flagModel->verifyParamsForAdd( $flagsToPages );
```

Comment:
I think that you could just use the Flags API to do the insert. Maybe create a private method wrapping the POST request?